### PR TITLE
Add snapshot filter support.

### PR DESCRIPTION
**Pull request by qiuyesuifeng**

### CODE REVIEW

File: server/balancer.go [hunk 1]
```diff
@@ -26,13 +26,20 @@ type Balancer interface {
...
+	// it will never be used as a from store.
 	minCapacityUsedRatio = 0.4
 	// If the used ratio of one storage is greater than this value,
-	// it will never be used as a selected target.
+	// it will never be used as a to store.
 	maxCapacityUsedRatio = 0.9
 
+	// If the sending snapshot count of one storage is greater than this value,
+	// it will never be used as a from store.
+	maxSnapSendingCount = uint32(5)
```
Comment (siddontang) [inline server/balancer.go]: I think 3 is ok, 5 seems a little large.

**Comment by siddontang** [comment]
Rest LGTM

**Comment by shenli** [comment]
LGTM

**Comment by shenli** [comment]
LGTM

**Pull request merged**